File: include/PolyWorksSDK/samples/plugins/common/VisualBasic.NET/VBNet3DSceneRendering/COM_SDK/Util/CComCommandParameter.vb
' =================================================================================================
' Class representing a COM command parameter
'
' Copyright © InnovMetric Logiciels Inc. 2012 All Rights Reserved
' -------------------------------------------------------------------------------------------------
Namespace VBNet3DSceneRendering

    ' =============================================================================
    ' Classe used to implement a COM parameter
    '
    ' Derive from IIMCommandParameter 
    ' -----------------------------------------------------------------------------
    <System.Runtime.InteropServices.ComVisible(True)> _
    <System.Runtime.InteropServices.ClassInterface(System.Runtime.InteropServices.ClassInterfaceType.None)> _
    Public Class CComCommandParameter
        Implements IMPluginLib.IIMCommandParameter

        ' =============================================================================
        '                           IIMCommandParameter Interface 
        ' =============================================================================

        ' =============================================================================
        ' See IIMCommandParameter.ParamDescGet
        ' -----------------------------------------------------------------------------
        Public Function ParamDescGet() As String _
        Implements IMPluginLib.IIMCommandParameter.ParamDescGet

            If Not m_paramInfo Is Nothing Then

                Return m_paramInfo.DescGet()

            End If

            Return String.Empty

        End Function

        ' =============================================================================
        ' See IIMCommandParameter.ParamPossibleValuesGet
        ' -----------------------------------------------------------------------------
        Public Function ParamPossibleValuesGet() As String _
        Implements IMPluginLib.IIMCommandParameter.ParamPossibleValuesGet

            If Not m_paramInfo Is Nothing Then

                Return m_paramInfo.ValuesGet()

            End If

            Return String.Empty

        End Function

        ' =============================================================================
        ' See IIMCommandParameter.ParamDirGet
        ' -----------------------------------------------------------------------------
        Public Function ParamDirGet() As IMPluginLib.EParameterDirection _
        Implements IMPluginLib.IIMCommandParameter.ParamDirGet

            If Not m_paramInfo Is Nothing Then

                Return m_paramInfo.DirGet()

            End If

            Return 0

        End Function

        ' =============================================================================
        ' See IIMCommandParameter.ParamTypeGet
        ' -----------------------------------------------------------------------------
        Public Function ParamTypeGet() As IMPluginLib.EParameterType _
        Implements IMPluginLib.IIMCommandParameter.ParamTypeGet

            If Not m_paramInfo Is Nothing Then

                Return m_paramInfo.TypeGet()

            End If

            Return 0

        End Function

        ' =============================================================================
        ' See IIMCommandParameter.ParamDefaultDoubleValueGet
        ' -----------------------------------------------------------------------------
        Public Function ParamDefaultDoubleValueGet(ByRef defaultValue_ As Double) As Integer _
        Implements IMPluginLib.IIMCommandParameter.ParamDefaultDoubleValueGet

            defaultValue_ = 0.0

            If Not m_paramInfo Is Nothing Then

                If m_paramInfo.HasDoubleDefaultValueGet() = True Then

                    defaultValue_ = m_paramInfo.DoubleDefaultValueGet()

                    Return 1

                End If

            End If

            Return 0

        End Function

        ' =============================================================================
        ' See IIMCommandParameter.ParamDefaultIntValueGet
        ' -----------------------------------------------------------------------------
        Public Function ParamDefaultIntValueGet(ByRef defaultValue_ As Integer) As Integer _
        Implements IMPluginLib.IIMCommandParameter.ParamDefaultIntValueGet

            defaultValue_ = 0

            If Not m_paramInfo Is Nothing Then

                If m_paramInfo.HasIntDefaultValueGet() = True Then

                    defaultValue_ = m_paramInfo.IntDefaultValueGet()

                    Return 1

                End If

            End If

            Return 0

        End Function

        ' =============================================================================
        ' See IIMCommandParameter.ParamDefaultStringValueGet
        ' -----------------------------------------------------------------------------
        Public Function ParamDefaultStringValueGet(ByRef defaultValue_ As String) As Integer _
        Implements IMPluginLib.IIMCommandParameter.ParamDefaultStringValueGet

            defaultValue_ = String.Empty

            If Not m_paramInfo Is Nothing Then

                If m_paramInfo.HasStringDefaultValueGet() = True Then

                    defaultValue_ = m_paramInfo.StringDefaultValueGet()

                    Return 1

                End If

            End If

            Return 0

        End Function

        ' =============================================================================
        '              CComCommandParameter specific methods
        ' =============================================================================

        ' =============================================================================
        ' Constructor
        '
        ' Parameter: info_ : Information used to construct parameter
        ' -----------------------------------------------------------------------------
        Public Sub New(ByVal info_ As CComCommandParameterInfo)

            m_paramInfo = New CComCommandParameterInfo(info_)

        End Sub

        Private m_paramInfo As CComCommandParameterInfo = Nothing

    End Class

End Namespace

File: include/PolyWorksSDK/samples/plugins/common/CSharp/CSharpCommands/COM_SDK/Util/CComCommandParameter.cs
// =================================================================================================
// Class representing a COM command parameter
//
// Copyright © InnovMetric Logiciels Inc. 2009 All Rights Reserved
// -------------------------------------------------------------------------------------------------
namespace IM
{
    // =============================================================================
    // Classe used to implement a COM parameter
    //
    // Derive from IIMCommandParameter 
    // -----------------------------------------------------------------------------
    [System.Runtime.InteropServices.ComVisible(true)]
    [System.Runtime.InteropServices.ClassInterface(System.Runtime.InteropServices.ClassInterfaceType.None)]
    public class CComCommandParameter : IMPluginLib.IIMCommandParameter
    {
        /* *****************************************************************************
        *                           IIMCommandParameter Interface 
        * -------------------------------------------------------------------------- */
        #region IIMCommandParameter

        // =============================================================================
        // See IIMCommandParameter.ParamDescGet
        // -----------------------------------------------------------------------------
        public string ParamDescGet()
        {
            if (m_paramInfo != null)
            {
                return m_paramInfo.DescGet();
            }
            return string.Empty;
        }

        // =============================================================================
        // See IIMCommandParameter.ParamPossibleValuesGet
        // -----------------------------------------------------------------------------
        public string ParamPossibleValuesGet()
        {
            if (m_paramInfo != null)
            {
                return m_paramInfo.ValuesGet();
            }
            return string.Empty;
        }

        // =============================================================================
        // See IIMCommandParameter.ParamDirGet
        // -----------------------------------------------------------------------------
        public IMPluginLib.EParameterDirection ParamDirGet()
        {
            if (m_paramInfo != null)
            {
                return m_paramInfo.DirGet();
            }
            return 0;
        }

        // =============================================================================
        // See IIMCommandParameter.ParamTypeGet
        // -----------------------------------------------------------------------------
        public IMPluginLib.EParameterType ParamTypeGet()
        {
            if (m_paramInfo != null)
            {
                return m_paramInfo.TypeGet();
            }
            return 0;
        }

        // =============================================================================
        // See IIMCommandParameter.ParamDefaultDoubleValueGet
        // -----------------------------------------------------------------------------
        public int ParamDefaultDoubleValueGet(out double defaultValue_)
        {
            defaultValue_ = 0.0;
            if (m_paramInfo != null)
            {
                if (m_paramInfo.HasDoubleDefaultValueGet())
                {
                    defaultValue_ = m_paramInfo.DoubleDefaultValueGet();
                    return 1;
                }
            }
            return 0;
        }

        // =============================================================================
        // See IIMCommandParameter.ParamDefaultIntValueGet
        // -----------------------------------------------------------------------------
        public int ParamDefaultIntValueGet(out int defaultValue_)
        {
            defaultValue_ = 0;
            if (m_paramInfo != null)
            {
                if (m_paramInfo.HasIntDefaultValueGet())
                {
                    defaultValue_ = m_paramInfo.IntDefaultValueGet();
                    return 1;
                }
            }
            return 0;
        }

        // =============================================================================
        // See IIMCommandParameter.ParamDefaultStringValueGet
        // -----------------------------------------------------------------------------
        public int ParamDefaultStringValueGet(out string defaultValue_)
        {
            defaultValue_ = string.Empty;
            if (m_paramInfo != null)
            {
                if (m_paramInfo.HasStringDefaultValueGet())
                {
                    defaultValue_ = m_paramInfo.StringDefaultValueGet();
                    return 1;
                }
            }
            return 0;
        }

        #endregion

        /* *****************************************************************************
         *              CComCommandParameter specific methods
        * -------------------------------------------------------------------------- */
        #region CComCommandParameter

        // =============================================================================
        // Constructor
        //
        // Parameter: info_ : Information used to construct parameter
        // -----------------------------------------------------------------------------
        public CComCommandParameter(CComCommandParameterInfo info_)
        {
            m_paramInfo = new CComCommandParameterInfo(info_);
        }

        private CComCommandParameterInfo m_paramInfo = null;

        #endregion
    }
}
